### PR TITLE
feat: add api as a service in docker-compose.yml

### PR DESCRIPTION
**Pull request by sergiomarrocoli**
I added this to the .env file in protectedplanet-api to get it to work
```
APPSIGNAL_LOG_PATH=/var/log/appsignal/protectedplanet-api.log
```

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -90,6 +90,25 @@ services:
...
+      - ${SSH_AUTH_SOCK}:${SSH_AUTH_SOCK}
+    network_mode: host
+    environment:
+      - SSH_AUTH_SOCK=${SSH_AUTH_SOCK}
+    env_file:
+      - ${API_ABSOLUTE_PATH}/.env
+    depends_on:
+      - db
+    stdin_open: true
+    tty: true
```
Comment (sergiomarrocoli) [inline docker-compose.yml:108]: not sure what these do

File: docker-compose.yml [hunk 2]
```diff
@@ -90,6 +90,25 @@ services:
...
+    build: ${API_ABSOLUTE_PATH}
+    command: /bin/bash -l -c "bundle exec rackup"
+    volumes:
+      - ${API_ABSOLUTE_PATH}:/ProtectedPlanetApi
+      - ${SSH_AUTH_SOCK}:${SSH_AUTH_SOCK}
+    network_mode: host
+    environment:
+      - SSH_AUTH_SOCK=${SSH_AUTH_SOCK}
+    env_file:
+      - ${API_ABSOLUTE_PATH}/.env
```
Comment (sergiomarrocoli) [inline docker-compose.yml:104]: what does this do?

File: docker-compose.yml (hunk 1 above)
Comment (pazolka) [reply docker-compose.yml:108]: needed for debugging, interactive console

File: docker-compose.yml (hunk 2 above)
Comment (pazolka) [reply docker-compose.yml:104]: mapping the ssh agent that was passed in the environmental variable right before `docker compose up` - not needed for the api service

Comment (pazolka) [reply docker-compose.yml:104]: both lines 99 and 102 were needed to make the ssh agent work inside the container

File: docker-compose.yml [hunk 3]
```diff
@@ -90,6 +90,25 @@ services:
       - protectedplanet_bundler:/usr/local/bundle
     network_mode: host
 
+  api:
+    container_name: protectedplanet-api
+    build: ${API_ABSOLUTE_PATH}
```
[the remark this answers is not in the document]
Comment (sergiomarrocoli) [reply docker-compose.yml:95]: i think it does if it's outside the context. will check again

**Review by pazolka — APPROVED** [review]

**Pull request merged**